pkg/detectors/plaidkey/plaidkey.go

```diff
@@ -59,38 +60,52 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 		uniqueIds[id] = struct{}{}
 	}
 
-	for key := range uniqueKeys {
-		for id := range uniqueIds {
+	for _, foundToken := range tokenPat.FindAllStringSubmatch(dataStr, -1) {
+		token := foundToken[1]
+		if detectors.StringShannonEntropy(token) < 3 {
+			continue
+		}
 
-			s1 := detectors.Result{
-				DetectorType: detectorspb.DetectorType_PlaidKey,
-				Raw:          []byte(key),
-			}
-			environments := []string{"sandbox", "production"}
-			if verify {
-				for _, env := range environments {
-					isVerified, _, verificationErr := verifyMatch(ctx, client, id, key, env)
-					s1.Verified = isVerified
-					s1.ExtraData = map[string]string{"environment": fmt.Sprintf("https://%s.plaid.com", env)}
-					s1.SetVerificationError(verificationErr, id, key)
+		uniqueTokens[token] = struct{}{}
+	}
+
+	for secret := range uniqueSecrets {
+		for id := range uniqueIds {
+			for token := range uniqueTokens {
+				s1 := detectors.Result{
+					DetectorType: detectorspb.DetectorType_PlaidKey,
+					Raw:          []byte(secret),
+					RawV2:        []byte(fmt.Sprintf(`%s:%s:%s`, secret, id, token)),
 				}
-				results = append(results, s1)
-				// if the environment is sandbox, we don't need to check production
-				if s1.Verified {
-					break
+				environments := []string{"sandbox", "production"}
```

Comment:
Since the environment name is embedded in the access token, why aren't we verifying it against the specific environment the token corresponds to, rather than checking against both?